internal/ccapi/service_instances.go

```diff
@@ -76,41 +61,89 @@ func (c CCAPI) GetServiceInstances(planGUIDs []string) ([]ServiceInstance, error
 		return nil, fmt.Errorf("no service_plan_guids specified")
 	}
 
-	var si serviceInstances
-	if err := c.requester.Get("v3/service_instances?"+BuildQueryParams(planGUIDs), &si); err != nil {
+	var receiver struct {
+		Instances []ServiceInstance `json:"resources"`
+		Included  struct {
+			Plans            []includedPlan            `json:"service_plans"`
+			Spaces           []includedSpace           `json:"spaces"`
+			Organizations    []includedOrganization    `json:"organizations"`
+			ServiceOfferings []includedServiceOffering `json:"service_offerings"`
+		} `json:"included"`
+	}
+
+	if err := c.requester.Get("v3/service_instances?"+BuildQueryParams(planGUIDs), &receiver); err != nil {
 		return nil, fmt.Errorf("error getting service instances: %s", err)
 	}
-	embedIncludes(si)
-	return si.Instances, nil
+
+	// Enrich with service plan and service offering data
+	servicePlanGUIDLookup := computeServicePlanGUIDLookup(receiver.Included.Plans, receiver.Included.ServiceOfferings)
+	for i := range receiver.Instances {
```

Comment:
suggestion: can we just merge the two loops? It is not clear why we need to iterate twice and there doesn't seem to be a reason, so probably better to just do the plan/offering and space/org look ups in the same loop.

Reply:
I was in two minds about this as I wasn't sure which was simplest. I've just made the change.